File: node/monitor.js
// An example MQTT service that publishes and subscribes
// Uses the AWS IoT messaging service
// Libraries
var aws_iot = require('aws-iot-device-sdk')

// Configuration
const service_name = 'test_monitor'
const aws_iot_config = {
  keyPath: "../certs/private.pem.key",
  certPath: "../certs/certificate.pem.crt",
  caPath: "../certs/root-CA.pem",
  clientId: service_name,
  host: "a267zn9knxsui0-ats.iot.eu-west-1.amazonaws.com",
  debug: false
}
const client = aws_iot.device(aws_iot_config)

// Main
client.on('connect', () => {
  // The '#' wildcard matches all topics below this one
  client.subscribe('workshop/#', function (err) {
    if (!err) {
      console.log('Monitoring topic workshop/#')
      console.log('Press Control-C to quit.')
    }
  })
})

client.on('message', (topic, message) => {
  console.log(`Message on topic: ${topic} is: ${message.toString()}`)
})

File: ruby/monitor.rb
# Imports
require 'aws_iot_device'

# Configuration
host = "a267zn9knxsui0-ats.iot.eu-west-1.amazonaws.com"
port = 8883
certificate_path = "../certs/certificate.pem.crt"
private_key_path = "../certs/private.pem.key"
root_ca_path = "../certs/root-CA.pem"
service_name = 'ruby_monitor'

# MQTT client
client = AwsIotDevice::MqttShadowClient::MqttManager.new(host: host, port: port, ssl: true, client_id: service_name)
client.config_ssl_context(root_ca_path, private_key_path, certificate_path)
client.connect

# Helpers
callback = Proc.new do |message|
  puts "On topic: #{message.topic}, message: #{message.payload}"
end

# Main
#
# Register a callback message logger on our topic
client.subscribe("workshop/#", 0, callback)

loop do
  sleep 3
end
